packages/office-ui-fabric-react/src/components/MarqueeSelection/MarqueeSelection.base.tsx

```diff
@@ -360,18 +387,53 @@ export class MarqueeSelectionBase extends BaseComponent<IMarqueeSelectionProps,
       }
     }
 
+    // set previousSelectedIndices to be all of the selected indices from last time
+    const previousSelectedIndices = this._allSelectedIndices;
+    this._allSelectedIndices = {};
+
+    // set all indices that is supposed to be selected in _allSelectedIndices
     for (const index in this._selectedIndicies!) {
       if (this._selectedIndicies!.hasOwnProperty(index)) {
-        selection.setIndexSelected(Number(index), true, false);
+        this._allSelectedIndices![index] = true;
       }
     }
 
     if (this._preservedIndicies) {
-      for (const index of this._preservedIndicies) {
-        selection.setIndexSelected(index, true, false);
+      for (const index of this._preservedIndicies!) {
+        this._allSelectedIndices![index] = true;
+      }
+    }
+
+    // check if needs to update selection, only when current _allSelectedIndices
+    // is different than previousSelectedIndices
+    let needToUpdate = true;
```

Comment:
Wait. Shouldn't this default to false?

Reply:
Please add a test to catch this bug. `needToUpdate` is never false.

Reply:
oh.. glad you caught that.  I was doing some tests on a previous PR iteration where I was forcing it to be true while diagnosing another problem, but forgot to set it back.  I am working on adding a test now.

Reply:
I have added a simple unit test that verifies that making a marquee around a selectable element will update the selection.  This will catch whether needToUpdate was properly set.